<launch>

    <!-- Arguments -->
    <arg name="robot_name" default="sam"/>

    <arg name="loop_freq" default="20."/>
    
    <!-- Topics -->
    <arg name="ref_topic_arg" default="/$(arg robot_name)/ctrl/ref"/>
    <arg name="setpoint_topic_arg" default="/$(arg robot_name)/ctrl/desired_setpoint"/>

    <!-- Nodes -->
    <node name="setpoint_node" pkg="depth_keeping" type="setpoint_node.py" output="screen">

        <param name="loop_freq" value="$(arg loop_freq)"/>

        <!-- Subscriber Topics -->
        <param name="setpoint_topic" value="$(arg setpoint_topic_arg)"/>

        <!-- Publisher Topics -->
        <param name="ref_topic" value="$(arg ref_topic_arg)"/>


    </node>
</launch>
